pkg/sqlutil/monitor.go

```diff
@@ -0,0 +1,147 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+const slowMsg = "SLOW SQL QUERY"
+
+// PromSQLQueryTime is exported temporarily while transitioning the core ORMs.
+var PromSQLQueryTime = promauto.NewHistogram(prometheus.HistogramOpts{
+	Name:    "sql_query_timeout_percent",
+	Help:    "SQL query time as a percentage of timeout.",
+	Buckets: []float64{10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120},
+})
+
+// MonitorHook returns a [QueryHook] that measures the timing of each query and logs about slow queries at increasing levels of severity.
+// When logAll returns true, every query and error will be logged.
+func MonitorHook(logAll func() bool) QueryHook {
```

Comment:
Could `MonitorHook` accept a straight `bool` arg?

Reply:
No, it is one of the few settings that can be changed at runtime, so we call this every time.